src/msg/async/ProtocolV2.cc

```diff
@@ -117,28 +88,114 @@ static void alloc_aligned_buffer(bufferlist &data, unsigned len, unsigned off) {
  * Protocol V2 Frame Structures
  **/
 
-// this type is used to specify the position of the signature within a frame
-struct signature_t {};
+static constexpr uint8_t CRYPTO_BLOCK_SIZE { 16 };
+
+using segment_t = ProtocolV2::segment_t;
+
+// V2 preamble is consisted with one or many preable blocks depending
+// on the number of segments particular frame needs. Each block holds
+// up to 2 segments and has its own CRC.
+//
+// XXX: currently the multi-segment facility is NOT implemented.
+struct preamble_block_t {
+  static constexpr std::size_t MAX_NUM_SEGMENTS = 4;
+
+  // ProtocolV2::Tag. For multi-segmented frames the value is the same
+  // between subsequent preamble blocks.
+  __u8 tag;
+
+  // Number of segments to go in entire frame. First preable block has
+  // set this to just #segments, second #segments - MAX_NUM_SEGMENTS,
+  // third to #segments - MAX_NUM_SEGMENTS and so on.
+  __u8 num_segments;
+
+  std::array<ProtocolV2::segment_t, MAX_NUM_SEGMENTS> segments;
+
+  uint8_t num_extra_preamble_blocks() const {
```

Comment:
This method is not used anywhere. Should we remove it, or will it be used when implementing the support for reading multiple preamble blocks?

Reply:
Yeah, it was for preamble chaining. We still have possibility to start doing that in the future but, at the moment, it's unnecessary as preamble became 32 bytes long. To be dropped.

Reply:
Dropped.